### PR TITLE
Skip testing when only tf or docs have changed

### PR DESCRIPTION
**Pull request by chronoslynx**
#### Details

Skip testing if and only if nothing impacting our code has changed.

Proof:

- ginkgo: https://app.circleci.com/pipelines/github/BishopFox/hawkeye/6427/workflows/fca83935-ff9f-43ca-901e-11e1c78cd2a5/jobs/18393
- go-component: https://app.circleci.com/pipelines/github/BishopFox/hawkeye/6427/workflows/fca83935-ff9f-43ca-901e-11e1c78cd2a5/jobs/18390

### CODE REVIEW

**Review by wkirschbaum-bishopfox — APPROVED** [review]

**Comment by chronoslynx** [comment]
`circleci step halt` doesn't quite do what I want, damn.

File: src/scripts/maybe-skip-testing.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+set -eu
+
+# Only files that have absolutely no bearing on our code should be listed here
+PAT='*\.(md|tf|svg|png)$|VERSION'
+
+set +o pipefail
+CHANGES=$(git diff --name-only HEAD~1 | grep -cvE "$PAT")
```
Comment (zirrostig) [inline src/scripts/maybe-skip-testing.sh:8]: why even bother with a count of changes? grep returns a non-zero exit when there are no matches.

**Review by zirrostig — APPROVED** [review]

File: src/scripts/maybe-skip-testing.sh (hunk 1 above)
Comment (zirrostig) [reply src/scripts/maybe-skip-testing.sh:8]: Why not copy the if statement from `src/scripts/ginkgo.sh` line 14?

File: src/scripts/ginkgo.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+run_ginkgo() {
+    ginkgo -r --randomizeAllSpecs --randomizeSuites "$@"
+}
+
+PACKAGE="${PACKAGE:-}"
+SKIP="${SKIP:-}"
+
+set +o pipefail
+if (git diff --name-only HEAD~1 | grep -cvE "${PAT}" >/dev/null); then
```
Comment (zirrostig) [inline src/scripts/ginkgo.sh:14]: The `-c` on grep is meaningless here.

Comment (chronoslynx) [reply src/scripts/ginkgo.sh:14]: Good catch, that was a change in the logic

**Pull request merged**